components/gfx/paint_context.rs

```diff
@@ -394,12 +419,34 @@ impl<'a> PaintContext<'a> {
         (Some(x1), Some(x2))
     }
 
-    fn intersect_ellipse_line(e: Ellipse, l: Line) -> (Option<Point2D<f32>>, Option<Point2D<f32>>) {
-        debug_assert!(l.end.x - l.start.x > f32::EPSILON, "Error line segment end.x > start.x!!");
-        // shift the origin to center of the ellipse.
-        let line = Line { start: l.start - e.origin,
-                          end: l.end - e.origin };
+    fn intersect_ellipse_line(_e: Ellipse, _l: Line) -> (Option<Point2D<f32>>,
+                                                         Option<Point2D<f32>>) {
+        let mut line = _l;
+        let mut e = _e;
+        let mut rotated_axises = false;
 
+        fn rotate_axises(point: Point2D<f32>, clockwise: bool) -> Point2D<f32> {
```

Comment:
Yes axes is right. I looked up the wrong plural, http://www.thefreedictionary.com/axises. Axises is the plural form of an animal.